### PR TITLE
Feature/issue 231 binary release workflow

### LINKED ISSUE #231: CI: create a new workflow to release the binary
**Describe the solution you'd like**
in each release build, we also automatically generate binary and then push to github asset.
at least we have to support(all are Little Endian)
1. Linux-x86_64
2. Linux-aarch64
3. Darwin-arm64
4. Darwin-x86_64

naming from `uname -s` + `uname -m`

**Describe alternatives you've considered**

**Additional context**
https://github.com/orgs/community/discussions/27009#discussioncomment-3254287
https://clickhouse.com/docs/en/development/build-cross-osx

### PR DESCRIPTION
**Pull request by leo-cai-timeplus**
PR checklist:
- Did you run ClangFormat ?
- Did you separate headers to a different section in existing community code base ?
- Did you surround `proton: starts/ends` for new code in existing community code base ?

Please write user-readable short description of the changes:
 a new workflow to build binary and release

### CODE REVIEW

File: .github/workflows/github_release.yml [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        # compiling
+        ./docker/packager/packager --package-type binary --docker-image-version clang-16 --build-type release --proton-build --enable-proton-local --cache ccache --ccache_dir $GITHUB_WORKSPACE/ccache --output-dir $GITHUB_WORKSPACE/output --compiler clang-16
+
+        if [ ! -f "$GITHUB_WORKSPACE/output/proton" ]; then
+          echo "Compiling proton Failed"
+          exit 127
+        fi
+
+        # upload binaries of v8
+        tar -zcf ./v8-cmake-x64.tar.gz -C $GITHUB_WORKSPACE/build_docker/contrib/v8-cmake/ bytecode_builtins_list_generator mksnapshot torque
```
Comment (yokofly) [inline .github/workflows/github_release.yml]: the pre-built binary does not need re-upload, only when v8 indeed needs re-built we compile and push it(like we modify the v8 code or the compiler toolchain change from clang-16 to clang-17).

File: .github/workflows/github_release.yml [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+        # compiling
+        ./docker/packager/packager --package-type binary --docker-image-version clang-16 --build-type release --proton-build --enable-proton-local --cache ccache --ccache_dir $GITHUB_WORKSPACE/ccache --output-dir $GITHUB_WORKSPACE/output --compiler clang-16-aarch64
+        if [ ! -f "$GITHUB_WORKSPACE/output/proton" ]; then
+          echo "Compiling proton Failed"
+          exit 127
+        fi
+
+        # upload binaries of v8
+        tar -zcf ./v8-cmake-arm64.tar.gz -C $GITHUB_WORKSPACE/build_docker/contrib/v8-cmake/ bytecode_builtins_list_generator mksnapshot torque
```
Comment (yokofly) [inline .github/workflows/github_release.yml]: same, no need upload i believe

File: .github/workflows/github_release.yml [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        run: |
+          # git login
+          gh auth login
+          gh auth status
+
+          # get proton tag
+          PROTON_TAG=$(git describe --abbrev=0)
+
+          # create a new release
+          gh release create $PROTON_TAG --generate-notes -d
```
Comment (yokofly) [inline .github/workflows/github_release.yml]: I prefer we add `--latest` and remember to remove the `-d --> --draft` this option when the final review down.

https://cli.github.com/manual/gh_release_create

File: .github/workflows/github_release.yml [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+      submodules: 'recursive'
+      run_mode: 'start' # start ec2 on demand instance
+      command: |
+        cd $GITHUB_WORKSPACE
+
+        # git config
+        git config user.name "proton-robot"
+        git config user.email "proton_robot@timeplus.io"
+
+        # download binary of v8
```
Comment (yokofly) [inline .github/workflows/github_release.yml]:
```suggestion
        # download the pre-built binary of v8 (this is only for cross-compile)
```

**Review by yokofly — CHANGES_REQUESTED** [review]

File: .github/workflows/github_release.yml (hunk 3 above)
Comment (leo-cai-timeplus) [reply .github/workflows/github_release.yml]: Yes, I 'll change it.
Do you think `--prerelease` should be added too?
Now all of our proton binary  releases are Pre-release.

Comment (yokofly) [reply .github/workflows/github_release.yml]: prelease -> not for production
latest-> latest
we first add the latest here.

File: .github/workflows/github_release.yml (hunk 4 above)
Comment (leo-cai-timeplus) [reply .github/workflows/github_release.yml]: Thanks, I'll alter the comment

File: .github/workflows/github_release.yml (hunk 3 above)
Comment (leo-cai-timeplus) [reply .github/workflows/github_release.yml]: ok.

File: .github/workflows/github_release.yml [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+        aws s3 cp --no-progress s3://tp-internal/proton/v8-cmake-x64.tar.gz $GITHUB_WORKSPACE/contrib/v8-cmake/
```
Comment (yokofly) [inline .github/workflows/github_release.yml]:
```suggestion
        aws s3 cp --no-progress s3://tp-internal/proton/cross-compile-prebuilt-binary/v8-cmake-x64.tar.gz $GITHUB_WORKSPACE/contrib/v8-cmake/
```

File: .github/workflows/release_build.yml [hunk 6]
```diff
@@ -154,3 +208,112 @@ jobs:
...
+      run_mode: 'start' # start ec2 on demand instance
+      command: |
+        cd $GITHUB_WORKSPACE
+
+        # git config
+        git config user.name "proton-robot"
+        git config user.email "proton_robot@timeplus.io"
+
+        # download the pre-built binary of v8 (this is only for cross-compile)
+        aws s3 cp --no-progress s3://tp-internal/proton/v8-cmake-x64.tar.gz $GITHUB_WORKSPACE/contrib/v8-cmake/
```
Comment (yokofly) [inline .github/workflows/release_build.yml]:
```suggestion
        aws s3 cp --no-progress s3://tp-internal/proton/cross-compile-prebuilt-binary/v8-cmake-x64.tar.gz $GITHUB_WORKSPACE/contrib/v8-cmake/
```

Comment (yokofly) [reply .github/workflows/release_build.yml]: @leo-cai-timeplus require change the folder path

**Review by yokofly — CHANGES_REQUESTED** [review]

File: .github/workflows/github_release.yml (hunk 5 above)
Comment (leo-cai-timeplus) [reply .github/workflows/github_release.yml]: Thanks

**Review by yokofly — APPROVED** [review]

**Pull request merged**